src/utils/getMarkdown.js

```diff
@@ -1,31 +1,35 @@
-import changeLinksAndImages from "./changeLinksAndImages";
+import changeLinksAndImages from './changeLinksAndImages';
 
-async function getMarkdown(match) {
+async function getMarkdown(match, history) {
   let link;
+  const { user, repository, path, file } = match.params;
 
   switch (match.path) {
-    case "/":
-      link = "https://api.github.com/repos/danilomacb/4noobs/contents/README.md";
+    case '/':
+      link =
+        'https://api.github.com/repos/danilomacb/4noobs/contents/README.md';
       break;
 
-    case "/:user/:repository":
-      link = `https://api.github.com/repos/${match.params.user}/${match.params.repository}/contents/README.md`;
+    case '/:user/:repository/:file/*':
+      link = `https://api.github.com/repos/${user}/${repository}/contents/4noobsDocs/${file}/${match.params[0]}`;
       break;
 
-    case "/:user/:repository/:file":
-      link = `https://api.github.com/repos/${match.params.user}/${match.params.repository}/contents/4noobsDocs/${match.params.file}`;
+    case '/:user/:repository':
+      link = `https://api.github.com/repos/${user}/${repository}/contents/README.md`;
       break;
 
     default:
       break;
   }
 
   let response = await fetch(link);
+  if (response.status === 404) return history.push('/');
```

Comment:
Eu discordo fortemente dessa alteração. Não acredito que faça sentido redirecionar se uma pagina não for encontrada. O ideal mesmo seria ter uma página de NOT FOUND, mas como não existe, acho que faz muito mais sentido abrir uma pagina em branco mesmo, pois fica mais facil de entender que essa pagina não existe dessa forma.

Reply:
Não achei legal deixar uma página em branco, mas já podemos ir preparando para a página de 404.